src/Interpreters/AsynchronousInsertQueue.cpp

```diff
@@ -168,17 +167,18 @@ AsynchronousInsertQueue::~AsynchronousInsertQueue()
         assert(dump_by_first_update_threads[i].joinable());
         dump_by_first_update_threads[i].join();
 
+        if (flush_on_shutdown)
+        {
+            for (auto & [_, elem] : shard.queue)
+                scheduleDataProcessingJob(elem.key, std::move(elem.data), getContext());
```

Comment:
pool could have a running jobs at a moment of shutdown, so technically, the is no difference if CH flushes queues by default when shutdown. So yes, let's do it by default.

Reply:
I agree, let's enable flush on shutdown by default.